src/kite/base.js

```diff
@@ -63,29 +71,31 @@ class Kite extends Emitter {
 
   setToken(token) {
     // FIXME: this setter is not symettrical with the getter
-    if (this.options.auth && this.options.auth === AuthType.token) {
+    const { auth } = this.options
+
+    if (!auth || auth === AuthType.token) {
```

Comment:
In the line 69, we are assigning to `this.options.auth` without making sure that it exists: `this.options.auth.key = token`, e.g it's gonna throw when auth is `null` or `undefined`. I just wanted to make sure we are checking for that as well. It's already throwing a `SyntaxError` with the current version, this just ensures that we are controlling the error behavior with minimal change.